### DISCUSSION TITLE
464 error between self-hosted Ztaidel behind ALB and Vercel

### DISCUSSION THREAD

**Discussion by TakumiHaruta** [zitadel/zitadel#5507] [Q&A]
Hi there,

We have self-hosted Zitadel instance built with Amazon ECS and ALB, and are trying to auth-request from the app on Vercel.
But the app got a 464 error from ALB when fetching data from `/.well-known/openid-configuration`. The app seems to be not able to access Zitadel at a very early stage.

464 error means that the protocol is not supported by the target group of ALB, and we know that the request protocol was `HTTP/1.1` which was not expected according to access logs of ALB.
The thing is Zitadel requires HTTP/2, so we don't know what is the workaround for this problem

We investigated several patterns;

| # | Client                          | Zitadel                       | success |
|---|---------------------------------|-------------------------------|---------|
| 1 | zitadel-nextjs on Vercel        | Amazon ECS behind ALB         | false   |
| 2 | zitadel-nextjs on local machine | Amazon ECS behind ALB         | false   |
| 3 | zitadel-nextjs on Vercel        | Only Amazon EC2 (without ALB) | true    |
| 4 | zitadel-nextjs on local machine | Only Amazon EC2 (without ALB) | true    |
| 5 | React SPA on local machine      | Amazon ECS behind ALB         | true    |
| 6 | zitadel-nextjs on Vercel        | Zitadel Cloud                 | true    |
| 7 | curl (http2)                    | Amaozn ECS behind ALB         | true    |

- Clients
    - https://github.com/zitadel/zitadel-nextjs
    - https://zitadel.com/docs/guides/start/quickstart#create-your-react-application-with-zitadel-oidc-pkce-authentication

Can anyone help us? Thanks

**Comment by TakumiHaruta** [comment]
Sorry it has been resolved.

There are some endpoints like `/.well-known/openid-configuration` which don't use HTTP/2.
It'd be helpful if there is a list for that.

**Reply by fforootd** [reply]
Since v2.22 we allow traffic also to pass to zitadel with http1.1 so it should be more smooth with some of the LB products.

https://github.com/zitadel/zitadel/pull/5376

**Reply by TakumiHaruta** [reply]
That sounds nice! I'll try test it then
